docs/data_structures/history_array.rst

```diff
@@ -7,25 +7,76 @@ history array
     H: numpy structured array
         History to store output from gen_f/sim_f/alloc_f for each entry
 
+Overview
+--------
+
 libEnsemble uses a NumPy structured array :ref:`H<datastruct-history-array>` to
-store output information about points generated by the ``gen_f`` and then the
-resultant output from the ``sim_f``. The history array also stores information
-about the time (since the epoch) when the points are generated/evaluated and
-which workers are/were involved.
-
-For ease, ``gen_f`` and ``sim_f`` are expected to return output as NumPy
-structured arrays. The names of the input fields for ``gen_f`` and ``sim_f``
-must be output from ``gen_f`` or ``sim_f``. In addition to output fields, the
-final history array returned by libEnsemble will include the following
-fields, almost all of which are protected from the user. These fields are
-populated internally:
+store information about each point (ensemble member) processed in the ensemble.
+Each entry contains multiple named fields, which can contain different data types,
+including scalar and arrays.
+
+The manager maintains the global history array for the ensemble. Each entry stores
+data generated by the ``gen_f`` and then the resultant output from the ``sim_f``,
+in addition to some reserved fields.
```

Comment:
Can we link to the reserved fields here?

Reply:
- [x] Added a referenced to reserved fields section.

Also made gen_f/sim_f references - though not sure whether should go to API (as I did) or the function writing guides eg. https://libensemble.readthedocs.io/en/docs-history_update/function_guides/simulator.html

Reply:
I think it is good.